pkg/controller/controller.go

```diff
@@ -145,6 +164,26 @@ func (c *controller) brokerDelete(obj interface{}) {
 	glog.V(4).Infof("Received delete event for Broker %v", broker.Name)
 }
 
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+// It enforces that the reconcileBrokerKey is never invoked concurrently with the same key.
+func (c *controller) brokerWorker() {
+	for {
+		func() {
```

Comment:
Yes, it is just a convenient way to make sure that `Done()` is always called on the dequeue key.

The deployment controller makes this anonymous function into a named one called `processNextWorkItem()` but they are doing the same thing
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go#L460-L476